--- a/bios/bios.c
+++ b/bios/bios.c
@@ -981,6 +981,7 @@
 			bios_reinitbyswitch();
 			bios_vectorset();
 			bios_screeninit();
+#if !defined(DISABLE_SOUND)
 			if (((pccore.model & PCMODELMASK) >= PCMODEL_VX) &&
 				(pccore.sound & 0x7e)) {
 				if(g_nSoundID == SOUNDID_MATE_X_PCM || ((g_nSoundID == SOUNDID_PC_9801_118 || g_nSoundID == SOUNDID_PC_9801_86_118 || g_nSoundID == SOUNDID_PC_9801_118_SB16 || g_nSoundID == SOUNDID_PC_9801_86_118_SB16) && np2cfg.snd118irqf == np2cfg.snd118irqp) || g_nSoundID == SOUNDID_PC_9801_86_WSS || g_nSoundID == SOUNDID_WAVESTAR || g_nSoundID == SOUNDID_PC_9801_86_WSS_SB16){
@@ -995,6 +996,7 @@
 					iocore_out8(0x18a, 0x3f);
 				}
 			}
+#endif
 			return(1);
 
 		case BIOS_BASE + BIOSOFST_09:
